### PR TITLE
[grppi] Add new port

### PR DESCRIPTION
**Pull request by NancyLi1013**
GrPPI is a header-only library. It doesn't currently support Intel compiler.
I try to add test as feature at first and find that there are many problems when building test.
Especially on Windows, except for some unsupported commands,` experimental/optional` cannot be found in many files. Even though I have updated them as `optional`, it will cause many other issues.
So I disable test now.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Related issue #5375.

File: ports/grppi/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: grppi
+Version: 0.4.0-1
```
Comment (cbezault) [inline ports/grppi/CONTROL]: Get rid of the `-1`

**Review by cbezault — CHANGES_REQUESTED** [review]
LGTM, just fix the version. We only add revision numbers when we need to.

File: ports/grppi/CONTROL (hunk 1 above)
Comment (NancyLi1013) [reply ports/grppi/CONTROL]: Thanks. Have updated.  I changed some code and so I updated the port version.

**Pull request merged**